src/Symfony/Component/DependencyInjection/Tests/EnvVarProcessorTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\EnvVarProcessor;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+
+class EnvVarProcessorTest extends TestCase
+{
+    const TEST_CONST = 'test';
+
+    public function testGetEnvString()
+    {
+        $container = new ContainerBuilder();
+        $container->setParameter('env(foo)', 'hello');
+        $container->compile();
+
+        $processor = new EnvVarProcessor($container);
+
+        $result = $processor->getEnv('string', 'foo', function () {
+            throw new LogicException('Shouldnt be called');
```

Comment:
Minor typo: `Shouldnt` -> `Shouldn\'t` or `Should not`

Reply:
Shouldn't we use `$this->fail()` instead?

Reply:
Yep, I couldn't remember what the function was called, I knew it existed